File: LeitorBarCode/app/src/main/java/com/dino/leitorbarcode/ScanCodeActivity.java
package com.dino.leitorbarcode;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;

import android.Manifest;
import android.content.pm.PackageManager;
import android.graphics.Camera;
import android.graphics.Matrix;
import android.os.Bundle;
import android.widget.Toast;

import com.google.zxing.BarcodeFormat;
import com.google.zxing.Result;

import me.dm7.barcodescanner.zxing.ZXingScannerView;

import static com.google.android.gms.common.util.CollectionUtils.listOf;

public class ScanCodeActivity extends AppCompatActivity implements ZXingScannerView.ResultHandler{

    ZXingScannerView ScannerView;
    int MY_PERMISSIONS_REQUEST_CAMERA=0;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ScannerView = new ZXingScannerView(this);
        setContentView(ScannerView);
    }

    @Override
    public void handleResult(Result result) {

        if (result != null){
            MainActivity.resultText.setText(result.getText().toString());
        }else {
            Toast.makeText(this, "Scan Canceldo", Toast.LENGTH_SHORT).show();
        }

        onBackPressed();
    }

    @Override
    protected void onPause() {
        super.onPause();
        ScannerView.stopCamera();
    }

    @Override
    protected void onResume() {
        super.onResume();
        ScannerView.setResultHandler(this);
        ScannerView.startCamera();
    }
    @Override
    protected void onPostResume() {
        super.onPostResume();
        if (ContextCompat.checkSelfPermission(getApplicationContext(), Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED) {
            ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.CAMERA}, MY_PERMISSIONS_REQUEST_CAMERA);
        }
        ScannerView.setResultHandler(this);
        ScannerView.startCamera();
    }
}

File: LeitordeBarCode/app/src/main/java/com/dino/leitordebarcode/ScanCodeActivity.kt
package com.dino.leitordebarcode

import android.Manifest
import android.app.ActionBar
import android.content.pm.PackageManager
import android.graphics.Camera
import android.os.Bundle
import android.view.ViewGroup
import android.view.ViewGroup.*
import android.view.ViewGroup.LayoutParams.*
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat
import com.google.zxing.BarcodeFormat
import com.google.zxing.Result
import me.dm7.barcodescanner.zxing.ZXingScannerView

class ScanCodeActivity : AppCompatActivity(), ZXingScannerView.ResultHandler{

    lateinit var ScannerView : ZXingScannerView
    var MY_PERMISSIONS_REQUEST_CAMERA = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        ScannerView = ZXingScannerView(this)
        setContentView(ScannerView)
    }

    override fun handleResult(p0: Result?) {

        if (p0 != null){
            MainActivity.resutado.text = p0?.text.toString()
        }else{
            Toast.makeText(applicationContext,"Scan Cancelado", Toast.LENGTH_LONG).show()
        }

        onBackPressed()
    }

    override fun onPause() {
        super.onPause()
        ScannerView.stopCamera()
    }

    override fun onResume() {
        super.onResume()
        ScannerView.setResultHandler(this)
        ScannerView.startCamera()

    }

    override fun onPostResume() {
        super.onPostResume()
        if (ContextCompat.checkSelfPermission(
                applicationContext,
                Manifest.permission.CAMERA
            ) != PackageManager.PERMISSION_GRANTED
        ) {
            ActivityCompat.requestPermissions(
                this,
                arrayOf(Manifest.permission.CAMERA),
                MY_PERMISSIONS_REQUEST_CAMERA
            )
        }
        ScannerView.setResultHandler(this)
        ScannerView.startCamera()
    }
}
